executor/runtime/docker/docker.go

```diff
@@ -947,35 +943,14 @@ func (r *DockerRuntime) Prepare(parentCtx context.Context) error { // nolint: go
 		return nil
 	})
 
-	if shouldStartSpectatord(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceSpectatord], &spectatordContainerName))
-	}
-
-	if shouldStartAtlasd(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceAtlasd], &atlasdContainerName))
-	}
-
-	if shouldStartMetatronSync(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceMetatron], &metatronContainerName))
-	}
-
-	if shouldStartSSHD(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceSshd], &sshdContainerName))
-	}
-
-	if shouldStartLogViewer(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceLogViewer], &logViewerContainerName))
-	}
-
-	if shouldStartServiceMesh(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceServiceMesh], &serviceMeshContainerName))
-	}
-
-	if shouldStartAbmetrix(&r.cfg, r.c) {
-		group.Go(r.createVolumeContainerFunc(sidecarConfigs[runtimeTypes.SidecarServiceAbMetrix], &abmetrixContainerName))
+	for sidecarName, sidecarConfig := range sidecarConfigs {
+		if sidecarConfig.Volumes != nil && sidecarConfig.EnabledCheck(&r.cfg, r.c) {
+			sidecarConfig.ContainerName = sidecarName
```

Comment:
Do you need `sidecarName` as an argument if `sidecarConfig` contains it?

Reply:
You are right, it is easier if the Config has the name in it.
I'll redo to be a list of sidcars instead of a map, so we can just pass around one sidecar object that "knows its name"